test/test_table.py

```diff
@@ -145,72 +168,208 @@ def test_column_retrieve_slow(self):
         db.refresh()
         db.wait_for_items()  # Avoid race condition
+        self.assertEqual(
+            [RecordedQuery('test_db', '.show database schema | project TableName, ColumnName, ColumnType | limit 10000')],
+            mock_kusto_client.recorded_queries,
+        )
         table = db.test_table
         self.assertIsInstance(table.foo, StringColumn)
         self.assertIsInstance(table.bar, NumberColumn)
         self.assertIsInstance(table['baz'], AnyTypeColumn)
         self.assertIsInstance(db['other_table']['foo'], AnyTypeColumn)
 
     def test_table_retrieve_error(self):
-        mock_kusto_client = MockKustoClient(tables_response=mock_tables_response([('test_table', [('foo', KustoType.STRING), ('bar', KustoType.INT)])]))
-        db = PyKustoClient(mock_kusto_client)['test_db']
+        mock_kusto_client = MockKustoClient(
+            tables_response=mock_tables_response([('test_table', [('foo', KustoType.STRING), ('bar', KustoType.INT)])]),
+            record_metadata=True,
+        )
+        db = PyKustoClient(mock_kusto_client, fetch_by_default=False)['test_db']
         db.refresh()
         db.wait_for_items()  # Avoid race condition
+        self.assertEqual(
+            [RecordedQuery('test_db', '.show database schema | project TableName, ColumnName, ColumnType | limit 10000')],
+            mock_kusto_client.recorded_queries,
+        )
         self.assertRaises(
             AttributeError("PyKustoClient(test_cluster.kusto.windows.net).Database(test_db) has no attribute 'test_table_1'"),
             lambda: db.test_table_1
         )
 
     def test_two_tables_retrieve(self):
-        mock_kusto_client = MockKustoClient(tables_response=mock_tables_response([
-            ('test_table_1', [('foo', KustoType.STRING), ('bar', KustoType.INT)]),
-            ('test_table_2', [('baz', KustoType.BOOL)])
-        ]))
-        db = PyKustoClient(mock_kusto_client)['test_db']
+        mock_kusto_client = MockKustoClient(
+            tables_response=mock_tables_response([
+                ('test_table_1', [('foo', KustoType.STRING), ('bar', KustoType.INT)]),
+                ('test_table_2', [('baz', KustoType.BOOL)])
+            ]),
+            record_metadata=True,
+        )
+        db = PyKustoClient(mock_kusto_client, fetch_by_default=False)['test_db']
         db.refresh()
         db.wait_for_items()  # Avoid race condition
+        self.assertEqual(
+            [RecordedQuery('test_db', '.show database schema | project TableName, ColumnName, ColumnType | limit 10000')],
+            mock_kusto_client.recorded_queries,
+        )
         self.assertIsInstance(db.test_table_1.foo, StringColumn)
         self.assertIsInstance(db.test_table_1.bar, NumberColumn)
         self.assertIsInstance(db.test_table_2['baz'], BooleanColumn)
         self.assertIsInstance(db['other_table']['foo'], AnyTypeColumn)
 
+    def test_union_retrieve(self):
```

Comment:
these tests have a lot of code duplications, I think it can be reduced quite easily.

Reply:
Consolidated tests to avoid duplication